scripts/build_chart_glyphs.py

```
@@ -0,0 +1,313 @@
+    url, sha = FONT_SOURCES[key]
+    CACHE.mkdir(parents=True, exist_ok=True)
+    dest = CACHE / f"{key}.ttf"
+    # Drop a corrupt cached copy (e.g. a download truncated by a dropped
+    # connection or a captive-portal HTML page) so we re-fetch instead of
+    # failing forever on a stale file.
+    if dest.exists() and _sha256(dest) != sha:
+        dest.unlink()
+    if not dest.exists():
+        print(f"  downloading {key} …")
+        tmp = dest.with_suffix(".ttf.part")
+        urllib.request.urlretrieve(url, tmp)
```

Ruff (0.15.18)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)